src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py

```diff
@@ -50,6 +50,16 @@ def create_domain(cmd, resource_group_name, hostname, contact_info, privacy=True
     web_client = web_client_factory(cmd.cli_ctx)
     hostname_availability = web_client.domains.check_availability(NameIdentifier(name=hostname))
 
+    if not hostname_availability.available:
+        raise CLIError("Custom domain name '{}' is not available. Please try again "
```

Comment:
please use error type in azclierror, e.g.
https://github.com/Azure/azure-cli/blob/3eea7fd773722b7a3c40b8115f29a3bad39b70bc/src/azure-cli-core/azure/cli/core/azclierror.py#L152-L154